Add nightmode

Xento opened the pull request:
I added a new mode to the base class "NIGHT".
Because Milight LEDs have a nightmode which means they light in white but with very little brightness.
Maybe it would be good to have the ability to add specialmodes to devices because they have a fourth mode "DISCO" with different effects.

Commit:
use async queue to only execute one command at the time

Commit:
Merge pull request #30 from philip1986/feature/improve_debounce_logic

use async queue to only execute one command at the time

Xento committed:
Add night mode

Implemented night mode for milight led.

Xento committed:
Merge pull request #8 from philip1986/develop

Update fork

Xento committed:
Fixed some bugs

Xento committed:
Optimizations and fixes

- Save the mode when set to white mode
- Only set color and brightnes if turned on

Xento committed:
Corrected mode if powered of

When you poweroff your led and it was in night mode, set it to the last
mode it was when it was turned on

Xento commented:
I will change the ci tests when you reviewed the code an you would merge it.

philip1986 commented:
I see two options:
1. we only support the "Night" mode Milight, but then I would argue you should not touch the base class with this PR and only attach the ActionProvider in case of Milight. 
1. We also start to support the Night mode for other devices (if its not supported by the device itself, we can just emulate it). In this case we could also add a night mode ability to the frontend. 

I personally would vote for the 2. solution, but @mwittig had once the plan to split the plugin by devices, than the 1. approach would be better.

So bottom line we will do it one way or the other. If you adjust the tests it would be nice if you also add some for the ActionHandler.

Xento commented:
Yeah I thought about the second solution.
I could add the emulation to the base class and every device which has an own night mode could overwrite it.

philip1986 commented:
Ok, do it like this :+1:

michbeck100 committed:
Added hyperion device

michbeck100 committed:
Added config schema, registering hyperion device, first try to connect to hyperion, not working yet.

michbeck100 committed:
Added hyperion-client and event-to-promise dependency.

michbeck100 committed:
Changing colors with hyperion works now. Turning on and off as well as setting brightness still need to be implemented.

michbeck100 committed:
Implemented brightness, turnOff and turnOn

michbeck100 committed:
Improved error handling

mwittig committed:
* package.json: updated dependencies, updated contributors list, added license property, added bugs property

mwittig committed:
* Moved "require 'blinkstick'" statement to constructor to circumvent exception in Hid.node on Windows.
* Fixed typos

mwittig committed:
0.2.0

mwittig committed:
Merge PR 33

mwittig committed:
Merge branch 'michbeck100-hyperion'

mwittig committed:
* Renamed device type definition LedLight to IwyMaster to be consistent with class name

mwittig committed:
* Fix for https://github.com/philip1986/pimatic-led-light/issues/27

mwittig committed:
* Revised and updated README

mwittig committed:
0.3.0

mwittig committed:
* added bluebird dependency as required for Hyperion

mwittig committed:
0.3.1

mwittig committed:
* Updated dependencies: iwy_master 0.2.3 which conatins fix for whte-mode https://github.com/philip1986/iwy_master/issues/2
* Fixed typos

mwittig committed:
0.3.2

mwittig committed:
* Updated vendor files to async 1.5 and spectrum 1.8. The latter resolves issue #39

mwittig committed:
* Implemented debounce-logic to avoid delay due to commands piling on bursts happening with brightness and color changes, in particular

Xento committed:
Merge pull request #9 from philip1986/master

Update to latest master version

Xento committed:
Merge pull request #10 from Xento/master

Update fork

Commit:
fix Hyperion sample config in README

Commit:
fix argument passing for setColor call

mwittig committed:
Reverted commit cc13952a171cd0138454dc686181844cc99192b6, see ticket #35

mwittig committed:
* Changed Milight driver parameter fir command elay and repeat to reduce lag, issue #35

mwittig committed:
Merge pull request #40 from philip1986/fix/loading_toast

fix argument passing for setColor call

mwittig committed:
0.3.3

Xento committed:
Merge pull request #11 from philip1986/master

Update to latest master version

Xento committed:
Fixed double sending brightness when changed from external or by zone 0

Xento committed:
Fixed strange bug

When the ColorActionProvider is the first one registered than the
ModeActionProvider doesn't work anymore.

Xento committed:
Removed NightMode

- removed NightMode
- made _updateState mode independent

Xento committed:
added compatibility with new base class

Xento committed:
Corrected tests

Xento committed:
only offer nightmode to devices which support it

Xento committed:
updated test

Xento committed:
Updated test

Xento committed:
fix

Xento committed:
Test

Xento commented:
I removed the nightmode from the base class.
Than I made the _updateState function more generic, so that it works even with an for the baseclass unknown mode.
My intention is that the device can add custom modes such as nightmode or disco without having to modify the baseclass.

Xento committed:
Removed setMode from base

Xento committed:
Updated version vom node-milight-rf24

Xento committed:
Merge pull request #12 from philip1986/master

Update

michbeck100 committed:
Changes for newer node version:
- Updated blinkstick dependency to 1.1.3
- fix for DummyLedLight

mwittig committed:
Merge pull request #57 from michbeck100/node_update

Support for node 4.x

mwittig committed:
Reverting to blinkstick@1.1.1 as 1.1.3 fails to build for node 0.10x

xCite1986 committed:
changed singlequotes to doublequotes in MilightRF24-example

xCite1986 committed:
Update README.md

mwittig committed:
Merge pull request #62 from xCite1986/master

changed singlequotes to doublequotes in MilightRF24-example

mwittig committed:
Temporary fix to resolve serialport build issues with pimatic 0.8 (node 0.10)

mwittig committed:
0.3.4

mwittig committed:
0.3.5

Xento committed:
Merge pull request #14 from Xento/fix-laststate

Xento committed:
Reverted to string as powerstate

Xento committed:
Merge remote-tracking branch 'refs/remotes/philip1986/master' into development

Xento committed:
Merge branch 'development' of https://github.com/Xento/pimatic-led-light into development

Xento committed:
Fixed device config schema
